pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -107,16 +111,39 @@ func (r *ReconcileUserAccount) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, err
 	}
 
-	var createdOrUpdated bool
-	var user *userv1.User
-	if user, createdOrUpdated, err = r.ensureUser(reqLogger, userAcc); err != nil || createdOrUpdated {
-		return reconcile.Result{}, err
-	}
+	// If the UserAccount has not been deleted, create or update user and identity resources.
+	// If the UserAccount has been deleted, delete secondary resources identity and user.
+	if userAcc.ObjectMeta.DeletionTimestamp.IsZero() {
+		// Add the finalizer if it is not present
+		if err := r.setFinalizers(userAcc); err != nil {
```

Comment:
minor suggestion: pass the finalizer name as an arg here, so we can "match" it with the call in line 132 (you could even use a constant for its name)